client/verta/verta/_cli/deployment/update.py

```diff
@@ -15,9 +17,40 @@ def update():
 @click.option("--run-id", "-r", help="Experiment Run to deploy.")
 @click.option("--strategy", "-s", type=click.Choice(['direct', 'canary'], case_sensitive=False), help="Strategy to use to roll out new deployment.")
 @click.option("--canary-rule", "-c", multiple=True, help="Rule to use for canary deployment. Can only be used alongside --strategy=canary")
+@click.option("--interval", "-i", type=click.IntRange(min=0), help="Rollout interval, in seconds. Can only be used alongside --strategy=canary")
+@click.option("--step", type=click.FloatRange(min=0.0, max=1.0), help="Rollout interval, in seconds. Can only be used alongside --strategy=canary")
 @click.option("--workspace", "-w", help="Workspace to use.")
 # TODO: more options
-def update_endpoint(path, run_id, strategy, canary_rule, workspace):
+def update_endpoint(path, run_id, strategy, canary_rule, interval, step, workspace):
     """List all endpoints available.
     """
-    raise NotImplementedError
+    if strategy != "canary" and (canary_rule or interval is not None or step is not None):
```

Comment:
`interval` and `step` can't be `0`, so I think we can do this to be less verbose. Maybe do the zero-check before.
```suggestion
    if strategy != "canary" and (canary_rule or interval or step):
```

Reply:
fixed.